aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -206,12 +206,7 @@ void FunctionalTensorWrapper::replace_(const Tensor& other) {
   value_ = other;
   // out= ops are allowed to resize the output tensors, mutating both the data and metadata of the tensor.
   // We need to propagate that metadata mutation to the wrapper (new size).
-  if (sizes() != value_.sizes() || strides() != value_.strides()) {
```

Comment:
I remember adding this check a while ago to fix a failing test, and I forget the reason (sigh). But I'm not too concerned if CI is green (and it makes sense to me not to want this for SymInts)